### PR TITLE
Update proof systems commit to include safe lagrange basis

### PR DESCRIPTION
**Pull request by 45930**
Explain your changes:
* Updates the proof systems submodule to point to https://github.com/o1-labs/proof-systems/commit/3689889cc414231e95d7827092679fe484f6aafb

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by volhovm** [comment]
!ci-build-me

**Comment by volhovm** [comment]
@45930 you'll have to update the bindings (`src/lib/crypto/kimchi_bindings/stubs` and probably also `..../wasm`), since the underlying proof-systems commit changes the SRS access interface.

**Comment by 45930** [comment]
!ci-build-me

**Comment by volhovm** [comment]
!ci-build-me

**Review by volhovm — APPROVED** [review]
Looks about right, feel free to merge. Accepted on the assumption that this was locally tested to work with o1js :) thanks!

File: src/lib/crypto/kimchi_bindings/wasm/src/srs.rs [hunk 1]
```diff
@@ -261,8 +253,8 @@ pub mod fp {
         domain_size: i32,
         i: i32,
     ) -> Option<WasmPolyComm> {
-        let bases = srs.0.lagrange_bases.get(&(domain_size as usize));
-        bases.map(|bases| bases[i as usize].clone().into())
+        let basis = srs.get_lagrange_basis_from_domain_size(domain_size as usize);
+        Some(basis[i as usize].clone().into())
```
Comment (mitschabaude) [inline src/lib/crypto/kimchi_bindings/wasm/src/srs.rs:257]: @45930 @mrmr1993 this change needs to be undone to unbreak Lagrange Basis caching and fix the performance regression. There was a reason this returned an option and wasn't simply doing the expensive computation

**Pull request merged**